hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/TestHddsDatanodeService.java

```diff
@@ -88,6 +125,44 @@ public void testStartup() throws IOException {
     service.close();
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = {OzoneConsts.SCHEMA_V1,
+      OzoneConsts.SCHEMA_V2, OzoneConsts.SCHEMA_V3})
+  public void testTmpDirOnShutdown(String schemaVersion) throws IOException {
+    ContainerTestVersionInfo.setTestSchemaVersion(schemaVersion, conf);
+    LOG.info("SchemaV3_enabled: " +
+        conf.get(DatanodeConfiguration.CONTAINER_SCHEMA_V3_ENABLED));
+    service.start(conf);
+
+    // Get volumeSet and store volumes in temp folders
+    // in order to access them after service.stop()
+    MutableVolumeSet volumeSet = service
+        .getDatanodeStateMachine().getContainer().getVolumeSet();
+    List<HddsVolume> volumes = StorageVolumeUtil.getHddsVolumesList(
+        volumeSet.getVolumesList());
+
+    for (HddsVolume volume : volumes) {
+      StorageVolumeUtil.checkVolume(volume, clusterId,
+          clusterId, conf, LOG, null);
+      // Create a container and move it under the tmp delete dir.
+      KeyValueContainer container = ContainerTestUtils
+          .setUpTestContainer(volume, clusterId, conf, schemaVersion);
+      assertTrue(container.getContainerFile().exists());
```

Comment:
For Schema_V3, an added entry into the db and deleting entry based on container in tmp directory would be a great addition.

Reply:
@neils-dev I've added the test, but it's on `TestEndPoint`. `TestHddsDatanodeService` tests tmp dir cleanup on shutdown where db is closed, so I added the check on startup where we don't have such issue.